api/rql/internal/predicate/expression/atom.go

```diff
@@ -0,0 +1,89 @@
+package expression
+
+import (
+	"time"
+
+	"github.com/puppetlabs/wash/api/rql"
+	"github.com/puppetlabs/wash/plugin"
+	"github.com/shopspring/decimal"
+)
+
+func Atom(p rql.ASTNode) rql.ASTNode {
```

Comment:
I'm not clear what an Atom is in this context.

Reply:
Hmm, not sure what you mean. Are you looking for a comment like

> Atom wraps p into a type that works with predicate expressions. The RQL will use this to call p's appropriate Eval* methods, where each Eval* method implements its interface-specific semantics. For example, if p is a Primary _and_ EntryPredicate, then the returned atom's EvalEntry method will be evaluated as a.EvalEntry(e) == p.EntryInDomain(e) && p.EvalEntry(e). Front-end interfaces to the RQL should _always_ use the Atom type when testing their parsed predicates to ensure correct evaluation semantics.

Note that I plan on exporting functions like `Atom`, `Not` in the `ast` package once I get to refactoring `find`'s parsing.

Reply:
Updated since I feel like that comment would be useful anyways

Reply:
I'm not sure what an atom is supposed to represent.

Reply:
Ok, I guess I kind of understand.